OpenTKAvalonia/BaseTkOpenGlControl.cs

```diff
@@ -17,81 +18,48 @@ public abstract class BaseTkOpenGlControl : OpenGlControlBase
     public AvaloniaKeyboardState KeyboardState = new();
 
     private AvaloniaTkContext? _avaloniaTkContext;
-    private FieldInfo _depthBufferField = typeof(OpenGlControlBase).GetField("_depthBuffer", BindingFlags.Instance | BindingFlags.NonPublic) ?? throw new("Unable to find _depthBuffer field");
-
-    public BaseTkOpenGlControl()
-    {
-        //Hook all Avalonia keyboard events to allow us to store Keyboard State 
-        InputManager.Instance!.PreProcess.Subscribe(e =>
-        {
-            if (e is not RawKeyEventArgs keyEvent)
-                return;
-
-            if (keyEvent.Type == RawKeyEventType.KeyDown)
-                KeyPressed(keyEvent);
-            else
-                KeyReleased(keyEvent);
-        });
-    }
-
 
     /// <summary>
     /// OpenTkRender is called once a frame to draw to the control.
     /// You can do anything you want here, but make sure you undo any configuration changes after, or you may get weirdness with other controls.
     /// </summary>
-    protected virtual void OpenTkRender()
-    {
-        
-    }
+    protected abstract void OpenTkRender();
 
     /// <summary>
     /// OpenTkInit is called once when the control is first created.
     /// At this point, the GL bindings are initialized and you can invoke GL functions.
     /// You could use this function to load and compile shaders, load textures, allocate buffers, etc.
     /// </summary>
-    protected virtual void OpenTkInit()
-    {
-        
-    }
+    protected abstract void OpenTkInit();
```

Comment:
Not sure why you've made this abstract? The whole point was that it didn't *have* to be overriden if you didn't want to.

Reply:
ah okay i just thought it would need to be overriden always anyways, i can revert that